daemon/k8s_watcher.go

```diff
@@ -409,9 +409,16 @@ func (d *Daemon) serviceAddFn(obj interface{}) {
 		return
 	}
 
+	log.Debugf("Adding k8s service %+v", svc)
+
+	nodePortSvcs := types.K8sServiceList{}
+
 	switch svc.Spec.Type {
-	case v1.ServiceTypeClusterIP, v1.ServiceTypeNodePort, v1.ServiceTypeLoadBalancer:
-		break
+	case v1.ServiceTypeNodePort, v1.ServiceTypeLoadBalancer:
+		nodePortSvcs = append(nodePortSvcs, types.NewK8sServiceInfo(nodeaddress.GetExternalIPv4()))
```

Comment:
Yes, the LB will happen to any endpoint which has an IPv4 address. So if one node maintains endpoints with IPv4 while others don't then all nodes will LB nodePort on all nodeIPs to the endpoints which have IPv4 addresses.